source/slang/slang-ir-spirv-legalize.cpp

```diff
@@ -363,9 +368,17 @@ struct SPIRVLegalizationContext : public SourceEmitterBase
         {
             format = decor->getFormat();
         }
+        if (textureType->GetBaseShape() == SLANG_TEXTURE_SUBPASS)
+        {
+            assert(format == ImageFormat::unknown);
+            return;
+        } 
+
+        // If the texture has no format decoration, try to infer it from the type.
         if (format == ImageFormat::unknown)
         {
-            // If the texture has no format decoration, try to infer it from the type.
+            // Do not infer format when shape is a subpass because SPRI-V spec defines
+            // subpass to have an unknown format 
```

Comment:
This comment may need to move.

Reply:
I will remove the comment since we have an error message slang throws in-place instead